codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules2/endpoint-provider-know-prop-override-class.java

```diff
@@ -179,6 +183,31 @@ private static RuleResult endpointRule11(QueryEndpointParams params, LocalState
         return RuleResult.error(locals.region() + " is not a valid HTTP host-label");
     }
 
+    private static RuleResult endpointRule12(QueryEndpointParams params, LocalState locals) {
+        if (params.useFipsEndpoint() == null && params.useDualStackEndpoint() != null && params.useDualStackEndpoint()
+            && params.arnList() != null) {
+            String firstArn = null;
+            RuleArn parsedArn = null;
+            if ((firstArn = RulesFunctions.listAccess(params.arnList(), 0)) != null) {
+                locals = locals.toBuilder().firstArn(firstArn).build();
+            } else {
+                return RuleResult.carryOn();
+            }
+            if ((parsedArn = RulesFunctions.awsParseArn(locals.firstArn())) != null) {
+                locals = locals.toBuilder().parsedArn(parsedArn).build();
```

Comment:
I was thinking more along the lines of 

```
            if ((firstArn = RulesFunctions.listAccess(params.arnList(), 0)) != null
                && (parsedArn = RulesFunctions.awsParseArn(firstArn)) != null) {
                locals = locals.toBuilder().parsedArn(parsedArn).firstArn(firstArn).build();
```

What you have works I think, just feeling uneasy about being able to reach the end of the second if with locals modified. That doesn't matter now, but before I would've never matter as the side effects would only happen in the if body scope.

Reply:
NOTE, No action item or change is expected, just leaving this as a side note.

Nesting could also work leaving the previous semantics, e.g.,

```java
        if ((firstArn = RulesFunctions.listAccess(params.arnList(), 0)) != null) {
                locals = locals.toBuilder().firstArn(firstArn).build();
                if ((parsedArn = RulesFunctions.awsParseArn(locals.firstArn())) != null) {
                    locals = locals.toBuilder().parsedArn(parsedArn).build();
                    return RuleResult.endpoint(Endpoint
                                                   .builder()
                                                   .url(URI.create("https://" + params.endpointId() + ".query."
                                                                   + locals.partitionResult().dualStackDnsSuffix()))
                                                   .putAttribute(
                                                       AwsEndpointAttribute.AUTH_SCHEMES,
                                                       Arrays.asList(SigV4aAuthScheme.builder().signingName("query")
                                                                                     .signingRegionSet(Arrays.asList("*")).build())).build());
                }
        }
        return RuleResult.carryOn();
```

Reply:
Yeah I also came up with that at first when handwriting it, just not sure how to modify the logic to do so. Need to plumb deeper through the call stack which is already pretty complicated